### PR TITLE
Update and refactor

### PR DESCRIPTION
**Pull request by teambailey**
Related task: [#328](https://app.zenhub.com/workspaces/front-end-6041273131e20d0011cb88dc/issues/polygon-io/frontend/328)

## Description
Changes to the available endpoints prompted this PR. Those changes include:
- Remove deprecated endpoints
- Add new endpoints
- Update existing response and query objects entries including associated types 
- Refactor existing client files
- Add Options to both the REST and WebSocket clients

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "keywords": [
     "polygon.io",
     "stock api",
+		"options api",
```
Comment (tharnach) [inline package.json:21]: I think we're mixing tabs and spaces here. I'd check other files in this PR as well.

File: src/rest/crypto/lastTrade.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  status?: string;
+	request_id?: string;
+  symbol?: string;
+  last?: {
+		conditions?: number[];
+		exchange?: number;
+		price?: number;
+		size?: number;
+		timestamp?: number;
+	}
```
Comment (tharnach) [inline src/rest/crypto/lastTrade.ts:15]: Some of these aren't optional, correct?

I'm seeing quite a few interfaces defined with optional properties. Are there cases in which the response doesn't have any of these properties?

Comment (teambailey) [reply src/rest/crypto/lastTrade.ts:15]: The spec has most response attributes listed as optional. Its pretty rare to have a required one.

And from how the spec reads, it looks like a completely empty response is a possibility

**Review by tharnach — APPROVED** [review]
Approved, but I'd like to get the opinion of someone who has a bit more experience with TypeScript to take a look as well.

File: src/rest/reference/optionsContracts.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  underlying_ticker?: string;
+  contract_type?: string;
+  expiration_date?:
+    | string
+    | {
+        lt?: string;
+        lte?: string;
+        gt?: string;
+        gte?: string;
+      };
```
Comment (clickingbuttons) [inline src/rest/reference/optionsContracts.ts:16]: :tada:

**Review by clickingbuttons — APPROVED** [review]
Types look fine assuming they match what our APIs really return. There might even be a way to autogenerate them from Swagger YML in the future.

**Review by mpatel18 — APPROVED** [review]
👍🏽

**Pull request merged**